src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue

```diff
@@ -50,58 +83,29 @@ export default {
       correct: true,
       code: null,
       contractSrc: null,
+      renderComponent: true,
+      activeItem: 0,
+      contractSrcHistory: [],
+      initSrcVersion: null,
+      preparedSource: [],
+      currentSrcTxId: null,
     };
   },
-  updated: function () {
-    Prism.highlightAll();
+  updated: async function () {
+    await Prism.highlightAll();
   },
-  async mounted() {
-    if (this.wasm) {
-      axios.get(`${this.gatewayUrl}/gateway/contract-source?id=${this.sourceId}`).then(async (fetchedSource) => {
-        if (!(fetchedSource.data.srcBinary instanceof Buffer)) {
-          fetchedSource.data.srcBinary = Buffer.from(fetchedSource.data.srcBinary.data);
-        }
-        const wasmSrc = new WasmSrc(fetchedSource.data.srcBinary);
-        let contractSrc;
-        try {
-          contractSrc = await wasmSrc.sourceCode();
-          console.log(contractSrc);
-        } catch (e) {
-          this.loaded = true;
-          this.correct = false;
-        }
-        let objFromContractSrc = Object.fromEntries(contractSrc);
 
-        if (fetchedSource.data.srcWasmLang == 'assemblyscript') {
-          this.contractSrc = this.getAs(objFromContractSrc);
-        } else if (fetchedSource.data.srcWasmLang == 'rust') {
-          this.contractSrc = this.getRust(objFromContractSrc);
-        } else if (fetchedSource.data.srcWasmLang == 'go') {
-          this.contractSrc = this.getGo(objFromContractSrc);
-        }
-        this.loaded = true;
-      });
-    } else {
-      // temporary until ArCode loads contracrt from the RedStone gateway
-      // if (this.isTestnet) {
-      axios.get(`${this.gatewayUrl}/gateway/contract-source?id=${this.sourceId}`).then((fetchedSource) => {
-        this.contractSrc = fetchedSource.data.src;
-        this.loaded = true;
+  async mounted() {
+    this.parseCode(this.source);
```

Comment:
something weird is happening here, parseCode is an async function and you are not awaiting it here. additionally - you are parsing code even though another source code will be displayed. cant we call `await this.parseCode(this.preparedSource[0]);` after `preparedSource` is created?

Reply:
fixed